FYDP_ModV1/src/version1.java

```diff
@@ -1,14 +1,181 @@
+            			 	case STRING:
+            			 		String cell = currCell.getStringCellValue();
+            			 		if(k == teacher_name_col) {
+                               	 teacherNames.add(currCell.getStringCellValue());
+                                }
+            			 		if (k == full_time_col) {
+            			 			
+            			 			if (cell.matches("(.*)x(.*)") ) {
+            			 				//assign full time
+            			 				fullTime = true;
+            			 				//fill with 1s
+            			 				int[] fullTimeTeacher = new int[30];
+            			 				for(int m = 0; m < 30; m++) {
+            			 					fullTimeTeacher[m]=1;
+            		
+            			 				}
+            			 				availableTime.add(fullTimeTeacher);
+            			 			}
+            			 		}
+            			 		if (k == french_certification_col) {
+            			 			if(cell.matches("(.*)x(.*)")) {
+            			 				if(first_french == true) {
+            			 					//Index of first french teacher 
+            			 					first_french_teacher=teacherNames.size();
+            			 					
+            			 					first_french = false;
+            			 				}
+            			 			}
+            			 		}
+            			 		break;
+            			 	case NUMERIC:
+            			 		break;
+            			 	case FORMULA:
+            			 		FormulaEvaluator evaluator = workbook.getCreationHelper().createFormulaEvaluator();
+            			 		CellValue cellValue = evaluator.evaluate(currCell);
+            			 		if(k==teacher_allocation_col ) {
+            			 			if(fullTime == false) {
+            			 			FTE.add(currCell.getNumericCellValue()/10);
+            			 			}else {
+            			 				FTE.add(1.0);
+            			 			}
+            			 		} 	
+            			 }
+            		}
+            		//collecting the available allocated timeslot for part time teachers
+            		int day1_col = 4;
+            		int day5_col = 8;
+            		if (fullTime == false) {
+            			int[] allocation_times = new int[30]; 
+            			int timeslot = 0;
+            			//iterate over column then row
+            			for(int col = day1_col;  col <= day5_col; col++) {
+            				for(int row = q; row < q+2 ; row++) {
+            					Row smallRow = sheetIndex.getRow(row);
+            					String m = smallRow.getCell(col).getStringCellValue();
+            					if (m.matches("(.*)x(.*)") ) {
+            						allocation_times[timeslot] = 1;
+            						allocation_times[timeslot+1] = 1;
+            						allocation_times[timeslot+2] = 1;
+            					}
+            					timeslot = timeslot+3;
+    
+            				}
+            			}
+            			availableTime.add(allocation_times);
+            		}
+            		
+            		q = q+2;
+            		teacherName = sheetIndex.getRow(q).getCell(teacher_name_col).getStringCellValue();
+
+            }
+           n2 = teacherNames.size();
+           int frenchTeach = n2 - first_french_teacher;
```

Comment:
I think this should change to 'int frenchTeach = n2 - first_french_teacher + 1; otherwise you exclude that 'first' french teacher from the count.... however, this only works if there is at least one french certified teacher. We can either assume there is always one french certified teacher or add in an if statement to calculate this variable based on if the 'french certification' column is populated